src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -46,6 +63,30 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
+  /** Stores the User in Datastore */
+  public void storeUser(String user) {
+    Entity userEntity = new Entity("User", user);
+    // Faster to compare ints than strings, use hash to check if user in Datastore
+    userEntity.setProperty("hashcode", user.hashCode());
+    if (userFound(user.hashCode()) != 0) {
+      datastore.put(userEntity);
+    }
+  }
+
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
So does this return the count with a max of 1000?  If so, note that in the comments.

Reply:
Noted in both user and message count, also updated limit to be 5000 for now